pkg/action/failover.go

```diff
@@ -66,6 +65,17 @@ func (ac *ActionController) validateBeforeFailover(action *storkv1.Action) {
 	// get sourceConfig from clusterPair in the destination cluster
 	sourceConfig, err := getClusterPairSchedulerConfig(migrationSchedule.Spec.Template.Spec.ClusterPair, migrationSchedule.Namespace)
 	if err != nil {
+		if *action.Spec.ActionParameter.FailoverParameter.SkipSourceOperations {
+			// if clusterPair is absent in destination cluster and skipSourceOperations is provided always move directly to ScaleUpDestination
+			msg := "Skipping source cluster operations and moving to the Failover ScaleUpDestinationStage since SkipSourceOperations is set to true"
+			logEvents := ac.printFunc(action, string(storkv1.ActionStatusSuccessful))
+			logEvents(msg, "out")
+			action.Status.Stage = storkv1.ActionStageScaleUpDestination
+			action.Status.Status = storkv1.ActionStatusInitial
+			action.Status.FinishTimestamp = metav1.Now()
+			ac.updateAction(action)
+			return
+		}
```

Comment:
If SkipSourceOperations & err == nil, do we want to go ahead and execute the rest of this function even if it's async-dr? Or should we make this check before SkipSourceOperations and return from there?

Reply:
Addressed. if async && skipSourceOperations we do not need to execute rest of the function